src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/DebugInstrument.java

```diff
@@ -0,0 +1,130 @@
+import com.oracle.truffle.js.runtime.builtins.JSFunction;
+import com.oracle.truffle.js.runtime.objects.JSObject;
+
+/**
+ * TruffleInstrument for the profiler
+ *
+ * @since 0.30
+ */
+@Registration(id = DebugInstrument.ID, name = "Debuging Instrument", version = "0.1", services = {DebugInstrument.class})
+public class DebugInstrument extends TruffleInstrument {
+    public static final String ID = "debugi";
+    private Instrumenter instrumenter;
+    private Env env;
+
+    public Env getEnv() {
+        return env;
+    }
+
+    public DebugInstrument() {
+        super();
+    }
+
+    @Override
+    protected OptionDescriptors getOptionDescriptors() {
+        return new OptionDescriptors() {
+
+            public Iterator<OptionDescriptor> iterator() {
+                return Arrays.asList(DebugInstrCLI.ods).iterator();
+            }
+
+            public OptionDescriptor get(String optionName) {
+                Iterator<OptionDescriptor> iter = iterator();
+                while (iter.hasNext()) {
+                    OptionDescriptor od = iter.next();
+                    if (od.getName().equals(optionName)) {
+                        return od;
+                    }
+                }
+                return null;
+            }
+        };
+    }
+
+    @Override
+    protected void onCreate(final Env env) {
+        this.env = env;
+        instrumenter = env.getInstrumenter();
+        env.registerService(this);
+        SourceSectionFilter sourceSectionFilter = SourceSectionFilter.newBuilder().tagIs(JSTags.ALL).build();
+        // What generates the input events to track?
+        SourceSectionFilter inputGeneratingObjects = SourceSectionFilter.newBuilder().tagIs(
+                        StandardTags.ExpressionTag.class,
+                        StandardTags.StatementTag.class,
+                        InputNodeTag.class).build();
+        env.getInstrumenter().attachExecutionEventFactory(sourceSectionFilter, inputGeneratingObjects, new ExecutionEventNodeFactory() {
+            public ExecutionEventNode create(EventContext context) {
+                // TODO Auto-generated method stub
+                return new ExecutionEventNode() {
+                    @Node.Child private InteropLibrary dispatch = InteropLibrary.getFactory().createDispatched(5);
+
+                    @Override
+                    public void onEnter(VirtualFrame frame) {
+                        /*
+                         * Internal sources are executed at engine startup time. Such sources
+                         * include internal code for the registration of builtins like Promise. We
+                         * skip all these internal events to ensure that tests are deterministic.
+                         */
+                        DynamicObject func = (DynamicObject) frame.getArguments()[1];
+                        try {
+                            dispatch.execute(JSFunction.createEmptyFunction(JSObject.getJSContext(func).getRealm()));
```

Comment:
@eleinadani This line of interop call results in unexpected output for the following test case:
```
(async()=>{let val = await 42; console.log(val);})()
console.log('###');
``` 
Expected:
```
###
42
```
After instrumentation:
```
```
42
###
```